src/core/thread/network_data_local.cpp

```diff
@@ -69,6 +68,44 @@ Error Local::AddOnMeshPrefix(const OnMeshPrefixConfig &aConfig)
     // of an IEEE 802.15.4 interface MUST have a length of 64 bits.
     VerifyOrExit(!aConfig.mSlaac || aConfig.mPrefix.mLength == OT_IP6_PREFIX_BITSIZE, error = kErrorInvalidArgs);
 
+    // Validate Prefix
+    VerifyOrExit((aConfig.GetPrefix().GetLength() > 0) && aConfig.GetPrefix().IsValid(), error = kErrorInvalidArgs);
+
+    // Validate Preference
+    switch (aConfig.mPreference)
+    {
+        case OT_ROUTE_PREFERENCE_LOW:
+        case OT_ROUTE_PREFERENCE_MED:
+        case OT_ROUTE_PREFERENCE_HIGH:
+            break;
+        default:
+            ExitNow(error = kErrorInvalidArgs);
+    }
+
+#if OPENTHREAD_FTD && OPENTHREAD_CONFIG_BACKBONE_ROUTER_ENABLE
+    if (aConfig.mDp)
+    {
+        // Check if domain prefix is already configured
+        OnMeshPrefixConfig prefixConfig;
+        VerifyOrExit(Get<BackboneRouter::Local>().GetDomainPrefix(prefixConfig) == kErrorNotFound, error = kErrorAlready);
```

Comment:
I think we should allow adding domain prefix when there is already one.

This is allowed in the original code:
```
node 1> prefix
2001:0:0:0::/64 parosD med 8c00
Done
node 1> prefix add 2002::/64 parosD
Done
node 1> prefix
2002:0:0:0::/64 parosD med fffe
Done
```

This would make buffer size checking for domain prefix more complex than a common prefix because the old domain prefix will be removed, thus leaving more space for the new domain prefix.

Reply:
@simonlingoogle There can be configured only one domain on-mesh prefix at a time. I have added a similar functionality to wpantund (https://github.com/openthread/wpantund/pull/489/files) to prohibit setting a new on-mesh domain prefix if there is already one. To add a new on-mesh domain prefix the old one must be explicitly removed. Which approach should I follow ?

Reply:
A little bit different from the other on-mesh prefixes, domain prefix is unique and designed to allow in-place change directly, meanwhile allow removing and then adding like the other prefixes.

One note is the domain prefix may only exist in bbr local, and not in local network data until bbr is enabled (currently bbr is enabled by default). If somehow bbr is disabled, you may not be able to check the existence of domain prefix configured via local network data.